libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -182,28 +180,32 @@ void EthStratumClient::disconnect()
 	m_responsetimer.cancel();
 	m_response_pending = false;
 
-	try {
+	if (m_socket && m_socket->is_open()) { 
+
+		try {
 		
-		boost::system::error_code sec;
+			boost::system::error_code sec;
 
-		if (m_conn.SecLevel() != SecureLevel::NONE) {
-			m_securesocket->shutdown(sec);
+			if (m_conn.SecLevel() != SecureLevel::NONE) {
+				m_securesocket->shutdown(sec);
+			}
+			else {
+				m_nonsecuresocket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, sec);
+			}
+
+			m_socket->close();
+			m_io_service.stop();
 		}
-		else {
-			m_nonsecuresocket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, sec);
+		catch (std::exception const& _e) {
+			cwarn << "Error while disconnecting:" << _e.what();
 		}
 
-		m_socket->close();
-		m_io_service.stop();
-	}
-	catch (std::exception const& _e) {
-		cwarn << "Error while disconnecting:" << _e.what();
+		if (m_securesocket) { m_securesocket = nullptr; }
```

Comment:
Maybe do it unconditionally? If it's null already nothing will change.

Reply:
It's just to simplify code. All 3 pointer are going to be nulled so it does no matter if any was a null before.